test/sandbox-test.js

```diff
@@ -2077,4 +2070,24 @@ describe("Sandbox", function() {
             assert.equals(object.prop, "bla");
         });
     });
+
+    describe(".assert", function() {
+        it("allows rebinding of .fail on a per-sandbox level", function() {
+            var sandboxA = createSandbox();
+            var sandboxB = createSandbox();
+
+            sandboxA.assert.failException = "CustomErrorA";
+            sandboxB.assert.failException = "CustomErrorB";
+
+            assert.exception(
+                function() {
+                    sandboxA.assert.fail("Some message");
+                },
+                { name: "CustomErrorA" }
+            );
+
+            sandboxA.restore();
+            sandboxB.restore();
```

Comment:
Not really seeing the point of sandboxB here, as it is not being exercised in any way. What purpose does it then serve? Should you not assert an exception on both sandboxes?

Reply:
This is just testing the behavior I demonstrated in the repro. If we have two tests running concurrently, one initializing A, the other B and *then* A makes an assertion, it uses its own assertion object, which is configured to throw the correct assertion rather than the global assertion object that may be reconfigured in B.

We can add in the second assertion, if you want, but this suffices and the test breaks without the changes

Reply:
Two months from now no one will remember the context for _why_ and will only look at the test implementation and go :thinking: .

If there is one part of the test that _should not throw_ (and does without the fix) or that _does throw_ (and should not) without the fix, then make it clear in the test. Wrap a `refute.exception(...)` around that block which should not block.

Reply:
btw refute is referee's "assert.not*"

Reply:
Ok. Please you add it. I get `Expected not to throw but threw 'CustomErrorA' ('Some message')` even though, I only want to refute that it throws `CustomErrorB`. I don't know referee at all.

Reply:
Ah, I'll fix. And I probably told you something wrong, anyway. I'll have to lookup those docs anyhow.